### PR TITLE
Make AccessCountDiskArray <: AbstractDiskArray + fix zip bug

### LINKED ISSUE #175: `zip` triggers lots of invalidations
We have a workload over at [ClimaAtmos.jl](https://github.com/CliMA/ClimaAtmos.jl), where we use SnoopCompile's [invalidations script](https://timholy.github.io/SnoopCompile.jl/dev/tutorial/#Cut-to-the-Chase:-A-copy-paste-analysis-of-invalidations), and the result is:

```
[ Info: 4296 methods invalidated for 178 functions (showing 10 functions)
┌─────────────────────────────────────────────────────────┬───────────────┬───────────────┬─────────────────┐
│ <file name>:<line number>                               │ Function Name │ Invalidations │ Invalidations % │
│                                                         │               │               │     (xᵢ/∑x)     │
├─────────────────────────────────────────────────────────┼───────────────┼───────────────┼─────────────────┤
│ DiskArrays/bZBJE/src/zip.jl:83                          │      zip      │     3247      │       14        │
│ DiskArrays/bZBJE/src/zip.jl:70                          │      zip      │     3247      │       14        │
│ DiskArrays/bZBJE/src/zip.jl:83                          │      zip      │     3247      │       14        │
│ FillArrays/eOEVm/src/fillbroadcast.jl:78                │  broadcasted  │     1220      │        5        │
│ CommonDataModel/G3moc/src/groupby.jl:267                │  broadcasted  │     1220      │        5        │
│ ClimaCore/ANgUC/src/MatrixFields/field_name_dict.jl:331 │  broadcasted  │     1220      │        5        │
│ ClimaCore/ANgUC/src/MatrixFields/field_name_dict.jl:475 │  broadcasted  │     1220      │        5        │
│ ClimaCore/ANgUC/src/MatrixFields/lazy_operators.jl:44   │  broadcasted  │     1220      │        5        │
│ GeometryBasics/ebXl0/src/offsetintegers.jl:40           │    convert    │      834      │        4        │
│ DiskArrays/bZBJE/src/generator.jl:53                    │   Generator   │      439      │        2        │
└─────────────────────────────────────────────────────────┴───────────────┴───────────────┴─────────────────┘
```
So, DiskArrays' [`zip`](https://github.com/meggart/DiskArrays.jl/blob/64f4de86ec3ab921759d7372c19766a463c58e61/src/zip.jl#L81-L83) seems to invalidate 9741 methods.

### PR DESCRIPTION
**Pull request by charleskawczynski**
I'm hoping that this will alleviate or close #175.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
Oh, wait, @rafaqz, can we just move the TestTypes module outside of source? Why would DiskArrays ship that anyway?

**Comment by felixcremer** [comment]
We use TestTypes in higher level packages  like YAXArrays to ensure that code is IO efficient.

**Comment by charleskawczynski** [comment]
> We use TestTypes in higher level packages like YAXArrays to ensure that code is IO efficient.

Couldn't that just be shipped as a library instead?

**Comment by felixcremer** [comment]
There might be a way to put this in an extension on Test, but then we would need to get the types out of the extension.

**Comment by charleskawczynski** [comment]
> There might be a way to put this in an extension on Test, but then we would need to get the types out of the extension.

Yeah, an extension would work, or even just make a DiskArraysTestTypes package in this repo, and then just depend on that where needed. I don't think that this is a blocker for this PR, but I think moving it to a separate package is a cleaner solution entirely.

File: src/util/testtypes.jl [hunk 1]
```diff
@@ -28,7 +28,7 @@ AccessCountDiskArray(a; chunksize=size(a),batchstrategy=DiskArrays.ChunkRead(Dis
 Base.size(a::AccessCountDiskArray) = size(a.parent)
 
 # Apply the all in one macro rather than inheriting
-DiskArrays.@implement_diskarray AccessCountDiskArray
+DiskArrays.@implement_diskarray_skip_zip AccessCountDiskArray
```
Comment (rafaqz) [inline src/util/testtypes.jl:31]:
```suggestion
```

This isn't needed, it's a subtype already

Comment (charleskawczynski) [reply src/util/testtypes.jl:31]: Ah, ok, thanks I'll update that

**Comment by charleskawczynski** [comment]
I think I found a bug!

```julia
function Base.zip(A1::AbstractDiskArray, A2::AbstractArray, As::AbstractArray...)
    return DiskZip(A1, A1, As...)
end
```
should be
```julia
function Base.zip(A1::AbstractDiskArray, A2::AbstractArray, As::AbstractArray...)
    return DiskZip(A1, A2, As...)
end
```
! This is probably pretty confusing since it was correctly implemented in `implement_zip`!

**Review by rafaqz — APPROVED** [review]

**Pull request merged**